### ISSUE TITLE
Change rules to type IDictionary<string, RateLimits>

### ISSUE THREAD

**Issue by BrightSoul** [stefanprodan/WebApiThrottle#37]
The ThrottlePolicy class currently exposes three sets of rules: ClientRules, EndpointRules and IPRules, all of the of type Dictionary<string, RateLimit>. This means that I have to load all of the rules in memory in advance and that is not desirable for scalability. 
What do you think of changing their type to IDictionary<string, RateLimit>, so the user could provide his own implementation of the dictionary? Rules could then be loaded when needed, from a remote key value store (for instance), and even change at runtime.

**Comment by stefanprodan** [comment]
Hello, please check the readme file, section: Update rate limits at runtime. You could make a pool request regarding IDictionary changes, I don't think will introduce any breaking changes.

**Comment by BrightSoul** [comment]
Hello Stefan, yes, I've made a pull request. Perhaps I've failed to explain what my intent was.
Suppose I have one million clients, each with his own client key. I don't necessarily want to load all of them beforehand in a Dictionary<string, RateLimits>.
Instead, as requests are coming, I'd like to go and query my storage (ie. Redis) for that particular key and only then load its RateLimits in memory.
If I'm able to provide my own implementation of IDictionary<string, RateLimits>, I can decide when Keys and Values are to appear in memory. Thanks for your attention ;)
